src/metabase/search/config.clj

```diff
@@ -264,17 +264,6 @@
   (conj default-columns :collection_id :collection_position :dataset_query :creator_id
         [:collection.name :collection_name]
         [:collection.authority_level :collection_authority_level]
-        [{:select   [:status]
-          :from     [:moderation_review]
-          :where    [:and
-                     [:= :moderated_item_type "card"]
-                     [:= :moderated_item_id :card.id]
-                     [:= :most_recent true]]
-          ;; order by and limit just in case a bug violates the invariant of only one most_recent. We don't want to
-          ;; error in this query
-          :order-by [[:id :desc]]
-          :limit    1}
-         :moderated_status]
```

Comment:
I think you still need `:moderated_status` here

Reply:
we don't need to, the `select-clause-for-model` will know it's missing and add one `[[:cast nil :text] :moderated_status]` for it, then at a later stage we'll call `replace-select` to find it and replace it with `:moderated_status` after we joined with the `moderation_review` table

Reply:
Got it. Can you add an explanation to the docstring for `columns-for-model` then, because that breaks the docstring as currently written https://github.com/metabase/metabase/blob/0c4a6d863f060e3636a7ae6ae8a948b09aca20dd/src/metabase/search/config.clj#L135

It seems to me like `columns-for-model` should be replaced by many calls to `replace-select` instead, so that there's no coupling between `columns-for-model` and `search-query-for-model` implementations.

e.g. a function like `add-collection-join-and-where-clauses` would also call `replace-select`

wdyt? not for this PR obviously

Reply:
I've updated the docstring here, wdyt? 97dd143db9 I'm not very good with the phrasing here

Reply:
Looks good to me